Argus-Labs/world-engine
# fix(nakama): Downgrade nakama dependencies to match internal nakama dependencies

## What is the purpose of the change

Software versions in shared objects must exactly match the versions found in the root project. These versions need to be downgraded to match the current Nakama docker container.

## Testing and Verifying

1) Run `go mod vendor` from `world-engine/relay/nakama`
2) cd to the starter-game-template directory
3) modify nakama.build to point to the `world-engine/relay/nakama` directory
4) run `mage nakama`

Without this PR, you'll see an error like this:

```
starter-game-template-nakama-1  | {"level":"fatal","ts":"2023-10-11T22:01:23.637Z","caller":"main.go:158","msg":"Failed initializing runtime modules","error":"plugin.Open(\"/nakama/data/modules/backend\"): plugin was built with a different version of package google.golang.org/protobuf/internal/pragma"}
starter-game-template-nakama-1 exited with code 1
```

With this PR, you should see:

```
starter-game-template-nakama-1  | {"level":"info","ts":"2023-10-11T22:03:26.864Z","caller":"main.go:189","msg":"Startup done"}
```
